docs/source/debugging.rst

```diff
@@ -0,0 +1,91 @@
+
+- ``AttributeError: 'str' object has no attribute 'sub’``
+    - this is caused by attempting to access an item in the solution dict with an invalid varkey
+
+- ``TypeError: unsupported operand type(s) for ** or pow(): 'int' and 'ParserHelper’``
+    - this normally indicates a ``Variable`` has been created with invalid units (i.e. units not supported by ``pint``)
+
+- ``RuntimeWarning: Primal solution violates constraint: 1.0000149786 is greater than 1``
+    - this warning may be seen in dual infeasible models, see *Dual Infeasible Models* below for more tips on debugging a dual infeasible model.
+
+- ``RuntimeWarning: Dual cost nan does not match primal cost 1.00122315152``
+    - this warning may be seen in dual infeasible models, see *Dual Infeasible Models* below for more tips on debugging a dual infeasible model.
+
+- ``RuntimeWarning: final status of solver 'cvxopt' was 'unknown', not 'optimal’`` or ``RuntimeWarning: final status of solver 'mosek' was ‘UNKNOWN’, not 'optimal’.``
+    - this is the most difficult warning to debug. It can be thrown when attempting to solve a dual infeasible model or a primal infeasible model. See *Dual Infeasible Models* and *Primal Infeasible Models* below for more information.
+
+- ``RuntimeWarning: final status of solver 'mosek' was 'DUAL_INFEAS_CER', not 'optimal’``
+    - this error is thrown when attempting to solve a dual infeasible model with MOSEK,  see *Dual Infeasible Models* below for more tips on debugging a dual infeasible model.
+
+
+Dual Feasible and Infeasible Models
+===================================
+
+Debugging
+---------
+
+In some cases a model will not solve because its variables are pushing to 0 or infinity. If the solver catches such behaviour it will return ``dual infeasible`` (or equivalent), but sometimes solvers do not catch it and return ``unknown``.
+
+``gpkit.constraints.bounded.BoundedConstraintSet`` is a
+simple tool that attempts to detect unbounded variables and get unbounded models to solve by adding extremely large upper bounds and extremely small lower bounds to all variables in a ConstraintSet.
+
+When a model with an BoundedConstraintSet is solved, it checks whether any variables slid off to the bounds, notes this in the solution dictionary and prints a warning (if verbosity is greater than 0).
+
+For example, Mosek returns ``DUAL_INFEAS_CER`` when attempting to solve the following model:
+
+.. literalinclude:: examples/unbounded.py
+
+Upon viewing the printed output,
+
+.. literalinclude:: examples/unbounded_output.txt
+
+it becomes clear that the problem is, unsurprisingly, that ``x`` is unbounded below in the original model.
+
+Details
+-------
+
+A dual feasible solution typically means that more than one unique solution meets the objective.   An example of a dual-feasible model is shown below. This model is dual-infeasible because there are multiple values of ``x`` and ``y`` that satisfy the constraint set and yield the globally optimum cost of 0.5.
```

Comment:
sorry, @mjburton11 and I both edited that and I think we had a mis-communication. will fix